### ISSUE TITLE
Convert extensions API errors from `anyhow` to a custom type

### ISSUE THREAD

**Issue by andreaphylum** [phylum-dev/cli#719]
Currently, the errors emitted by the Deno ops for the extensions API are `anyhow::Error`. We should switch to a custom type in order to provide users with more actionable and structured information.

**Comment by cd-work** [comment]
I rewrote the API module to use a custom error type [here](https://github.com/phylum-dev/cli/tree/api_errors), but then realized that the Deno API enforces `anyhow::Error` to be the error type.

I've checked how errors are submitted to JavaScript by logging the error and it seems like the `Display` implementation of whatever is passed to `anyhow` is used to convert the error to a String.

As a result, I don't think it's possible for us to easily resolve this issue without making **significant** upstream changes that are not unlikely to be denied.

So rather than going forward with a custom error type, I'd be curious if someone had alternative ideas for solving this issue that would stick with a string-based error type already supported by Deno.
